jwst/outlier_detection/outlier_detection_step.py

```diff
@@ -96,11 +95,18 @@ def _build_reffile_container(self, reftype):
         reffile_to_model = {'gain': datamodels.GainModel,
             'readnoise': datamodels.ReadnoiseModel}
 
-        reffiles = [self.get_reference_file(im, reftype) for im in self.input_models]
+        if reftype in self.input_models[0].meta.ref_file._instance:
+            reffile_names = [tm.meta.ref_file._instance[reftype]['name'] for tm in self.input_models]
+            if len(set(reffile_names)) == 1:
+               reffiles = [self.get_reference_file(self.input_models[0],reftype)]*len(reffile_names)
+            else:
+                reffiles = [self.get_reference_file(im, reftype) for im in self.input_models]                 
```

Comment:
So it looks like the call to get_reference_file, which invokes CRDS, is still being made even when the ref file name was already available from the meta data in the input model(s)? From what I can tell, the only increase in efficiency is in the case where all the input models had the same ref file name in the meta data, in which case only 1 call is made to get_reference_files (and hence CRDS), rather than calling it for every input.

Why not just use the ref file name(s) retrieved from tm.meta.ref_file._instance[reftype]['name'] in line 99 and forget about calling get_reference_file altogether? Only resort to calling get_reference_file when the names were not available in the meta data (i.e. as is done in line 105).

All get_reference_file is doing is returning the file **_name_** for you, which you already have in the meta data. It doesn't actually get (open and load) the reference file. You have to do that using a datamodels.open() call of some sort (usually invoking the appropriate data model type, such as datamodels.GainModel() or datamodels.ReadNoiseModel()).